fission/function.go

```diff
@@ -69,40 +74,87 @@ func printPodLogs(c *cli.Context) error {
 	return nil
 }
 
-func getInvokeStrategy(minScale int, maxScale int, executorType string, targetcpu int) fission.InvokeStrategy {
-
-	if maxScale == 0 {
-		maxScale = 1
-	}
+func getInvokeStrategy(c *cli.Context, existingInvokeStrategy *fission.InvokeStrategy) (strategy *fission.InvokeStrategy, err error) {
 
-	if minScale > maxScale {
-		log.Fatal("Maxscale must be higher than or equal to minscale")
-	}
+	var fnExecutor, newFnExecutor fission.ExecutorType
 
-	var fnExecutor fission.ExecutorType
-	switch executorType {
+	switch c.String("executortype") {
 	case "":
-		fnExecutor = fission.ExecutorTypePoolmgr
+		fallthrough
 	case fission.ExecutorTypePoolmgr:
-		fnExecutor = fission.ExecutorTypePoolmgr
+		newFnExecutor = fission.ExecutorTypePoolmgr
 	case fission.ExecutorTypeNewdeploy:
-		fnExecutor = fission.ExecutorTypeNewdeploy
+		newFnExecutor = fission.ExecutorTypeNewdeploy
 	default:
-		log.Fatal("Executor type must be one of 'poolmgr' or 'newdeploy', defaults to 'poolmgr'")
-	}
-
-	// Right now a simple single case strategy implementation
-	// This will potentially get more sophisticated once we have more strategies in place
-	strategy := fission.InvokeStrategy{
-		StrategyType: fission.StrategyTypeExecution,
-		ExecutionStrategy: fission.ExecutionStrategy{
-			ExecutorType:     fnExecutor,
-			MinScale:         minScale,
-			MaxScale:         maxScale,
-			TargetCPUPercent: targetcpu,
-		},
+		return nil, errors.New("Executor type must be one of 'poolmgr' or 'newdeploy', defaults to 'poolmgr'")
+	}
+
+	if existingInvokeStrategy != nil {
+		fnExecutor = existingInvokeStrategy.ExecutionStrategy.ExecutorType
+
+		// override the executor type if user specified a new executor type
+		if c.IsSet("executortype") {
+			fnExecutor = newFnExecutor
+		}
+	} else {
+		fnExecutor = newFnExecutor
+	}
+
+	if fnExecutor == fission.ExecutorTypePoolmgr {
+		if c.IsSet("mincpu") || c.IsSet("maxcpu") || c.IsSet("minmemory") || c.IsSet("maxmemory") {
+			log.Warn("To limit CPU/Memory for function with executor type \"poolmgr\", please specify resources limits when creating environment")
+		}
+		strategy = &fission.InvokeStrategy{
+			StrategyType: fission.StrategyTypeExecution,
+			ExecutionStrategy: fission.ExecutionStrategy{
+				ExecutorType: fission.ExecutorTypePoolmgr,
+			},
+		}
+	} else {
+		// set default value
+		targetCPU := DEFAULT_TARGET_CPU_PERCENTAGE
+		minScale := DEFAULT_MIN_SCALE
```

Comment:
Ok, I see, just making sure I understand:

1. If user sets nothing except executor=newdeploy, minscale=1
2. if users sets executor=newdeploy and minscale=0, they get minscale=0

And if executor is not set, it defaults to poolmgr.

So I was wondering: if minscale=1, and executor is not specified, should we default executor=newdeploy?  Or is that too confusing?

Reply:
no we should not do that, its too confusing. I add a warning message here to tell users to specify executor type explicitly instead.
https://github.com/fission/fission/pull/970/files#diff-17e3b56f4e0c4afc96fc4653126df7f8R105